### PR TITLE
 Reformat and Edit Enumerable Lesson

### PR DESCRIPTION
**Pull request by KevinMulhern**
* Add simple explanation of objects article to Basic Data Types Lesson
* Extract advanced topics to a future lesson
* Change some examples for clarity
* Format code examples for Kramdown
* Add assignment section

### CODE REVIEW

**Comment by leosoaivan** [comment]
Agree completely. This lesson is too long. 4 of each sounds perfect.

File: new_course/1_programming_fundementals/basic_enumerable_methods.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ### Introduction
-
-In Ruby, `Enumerable` is a module that provides a set of methods that are available to collection classes, such as `Array` or `Hash`. The basic requirement to use enumerable methods is that the collection class in question must have an `#each` method (further explained below). These enumerable methods are handy for searching or looping through a collection, and are among the most important tools for Rubyists.
+In Ruby, enumerables are a set of methods that are available to collection objects, such as arrays or hashes. These enumerable methods are handy for searching or looping through a collection, and are among the most important tools for Rubyists, they are a large part of what makes Ruby such a joy to work with.
```
Comment (leosoaivan) [inline new_course/1_programming_fundementals/basic_enumerable_methods.md]: I'd break up the last sentence, maybe at the second comma.

**Comment by KevinMulhern** [comment]
This is ready for a re-review when you get the time please @leosoaivan @leosoaivan

**Comment by leosoaivan** [comment]
Great stuff! @leila-alderman may want to give it a finer pass-through.

**Pull request merged**